tests/cases/compiler/spellingSuggestionGlobal1.ts

```diff
@@ -0,0 +1,3 @@
+export {}
+declare global { const x: any }
+global.x // should not suggest `global` (GH#42209)
```

Comment:
In this case, it should probably suggest `globalThis`.

Reply:
Did you miss [my reply](https://github.com/microsoft/TypeScript/issues/42209#issuecomment-754903122) after you suggested that on the issue?

[`globalThis.x` is not valid here](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBAbwL4CgAmwDGAbAhlYOAc2wgCNdtE5MIA7AZ3hAC45c6BPOVE8ygCoALAJYMAdCCA).

Reply:
It is valid when you do:
```ts
declare global { var x: any }
```

Reply:
Let's make sure we're on the same page:

**Are you saying `spellingSuggestionGlobal1.ts` (as written, with `const x`) should produce the error message `Cannot find name 'global'. Did you mean 'globalThis'?`?**

If so, I disagree. Users will follow that suggestion and change `global.x` to `globalThis.x`, then get another error (`Property 'x' does not exist on type 'typeof globalThis'.`), then be confused/frustrated and have to research why it doesn't work, then probably end up ignoring the suggestion anyway and write `x` instead of `globalThis.x`.

**Or are you saying that in situations where `x` is a known property of `globalThis` and the user writes `global.x` that it would be nice to say `Cannot find name 'global'. Did you mean 'globalThis'?`?**

If so, I agree, but unfortunately that would be tricky to implement (unless I'm missing something). I'd bet that this situation occurs rarely in the wild, so it might not be worth the complexity cost to maintain a dedicated suggestion for it. I could be wrong, though.

In any case it seems orthogonal to the bugfix in this pull request. Currently [if you write something like `global.Math` the error message won't suggest `globalThis`](https://www.typescriptlang.org/play?#code/OYGw9gRghiB0CyUAuALIA), either.

---

Regardless, I should add a test case that uses `var` in the global declaration. I'll push a commit for that soon.